engine-src/Game/LambdaHack/Client/UI/HandleHumanGlobalM.hs

```diff
@@ -1043,35 +1042,47 @@ closeDirHuman = do
   leader <- getLeaderUI
   b <- getsState $ getActorBody leader
   lvl <- getLevel $ blid b
-  let vPts = vicinityUnsafe $ bpos b 
+  let vPts = vicinityUnsafe $ bpos b
       openPts = filter (Tile.isClosable cotile . at lvl) vPts
-  case openPts of 
+  case openPts of
     []  -> failSer CloseNothing
     [o] -> closeTileAtPos $ Just o
-    _   -> pickPoint "close" >>= closeTileAtPos 
-   
+    _   -> pickPoint "close" >>= closeTileAtPos
+
 -- | Close tile at given position.
 closeTileAtPos :: MonadClientUI m
                => Maybe Point -> m (FailOrCmd RequestTimed)
 closeTileAtPos mp = case mp of
   Nothing -> failWith "never mind"
   Just p -> do
-    COps{cotile} <- getsState scops
+    COps{cotile, coTileSpeedup} <- getsState scops
     leader <- getLeaderUI
     b <- getsState $ getActorBody leader
-    if (p `chessDist` bpos b == 1) then do
-      actorSk <- leaderSkillsClientUI
-      lvl <- getLevel $ blid b
-      let alterSkill = Ability.getSk Ability.SkAlter actorSk
-      if | Tile.isOpenable cotile $ lvl `at` p 
-          -> failSer CloseClosed
-         | alterSkill <= 1
-          -> failSer AlterUnskilled
-         | otherwise 
-          -> do
-             msgAddDone p "close"
-             return $ Right $ ReqAlter p 
-    else failSer CloseDistant 
+    actorSk <- leaderSkillsClientUI
+    lvl <- getLevel $ blid b
+    let alterSkill = Ability.getSk Ability.SkAlter actorSk
+        t = lvl `at` p
+        isOpen = Tile.isClosable cotile t
+        isClosed = Tile.isOpenable cotile t
+        isModifiable = Tile.isModifiable coTileSpeedup t
+    case (isModifiable, isClosed, isOpen) of
+      (False, _, _) -> failSer CloseNothing
+      (True, False, False) -> failSer CloseNonClosable
+      (True, True,  False) -> failSer CloseClosed
+      (True, True,  True) -> failSer TileOpenClosed
```

Comment:
I suspect this case is not possible:

https://github.com/LambdaHack/LambdaHack/blob/master/definition-src/Game/LambdaHack/Content/TileKind.hs#L121

If so, please just call `error`, citing the cause and pointing at "TileKind content validation".